app/mci/cmd/create.go

```diff
@@ -173,25 +178,36 @@ func getClusters(kubeconfig string) ([]string, error) {
 	if err != nil {
 		return nil, fmt.Errorf("error in getting contexts from kubeconfig: %s", err)
 	}
+	glog.V(5).Infof("Contexts found:\n%v", output)
 	return strings.Split(output, "\n"), nil
 }
 
 // Creates the given ingress in the given list of clusters.
 func createIngressInClusters(kubeconfig, ingressFilename string, clusters []string) error {
-	kubectlArgs := []string{"kubectl"}
-	if kubeconfig != "" {
-		kubectlArgs = append(kubectlArgs, fmt.Sprintf("--kubeconfig=%s", kubeconfig))
+	var ing v1beta1.Ingress
+	if err := unmarshall(ingressFilename, &ing); err != nil {
```

Comment:
RunCreate already unmarshalls the object.
We can update it to pass the unmarshalled ingress object to createIngress instead of passing the ingress file name.
We were passing file name before since that is what kubectl wanted.